packages/apps/transfer/src/services/transfer/coin-transfer.ts

```diff
@@ -77,3 +77,94 @@ export async function transferCreate(
   await pactCommand.send(generateApiHost(networkId, chainId));
   return pactCommand;
 }
+
+export async function safeTransferCreate(
+  fromAccount: string,
+  toAccount: string,
+  amount: string,
+  fromPrivateKey: string,
+  toPrivateKey: string,
+  chainId: ChainId,
+  networkId: ChainwebNetworkId,
+): Promise<PactCommand> {
+  if (isNaN(Number(amount))) {
+    throw new Error('Amount must be a number');
+  }
+
+  // sender need to send extra coins to the receiver which will be returned by the receiver
+  const amountWithExtra = Number(amount) + 0.000000000001;
+
+  const pactCommand = new PactCommand();
+  pactCommand.code = `(coin.transfer-create "${fromAccount}" "${toAccount}" (read-keyset "ks") ${convertDecimal(
+    amountWithExtra.toString(),
+  )})(coin.transfer "${toAccount}" "${fromAccount}" ${convertDecimal(
+    '0.000000000001',
+  )})`;
+
+  pactCommand
+    .addCap('coin.GAS', onlyKey(fromAccount))
+    .addCap(
```

Comment:
Is it on purpose that only `x:y` type accounts are allowed? There can be named accounts that don't have the prefix `x:` (`x` being anything)